wire-moshi-support/src/test/java/com/squareup/wire/MoshiTest.java

```diff
@@ -0,0 +1,298 @@
+        .ext_opt_sint64(Long.MIN_VALUE / 2 + 178)
+        .ext_opt_bool(true)
+        .ext_opt_float(1.2345e6F)
+        .ext_opt_double(1.2345e67)
+        .ext_opt_nested_enum(AllTypes.NestedEnum.A)
+        .ext_opt_nested_message(nestedMessage)
+        .ext_rep_int32(list(Integer.MAX_VALUE))
+        .ext_rep_uint64(list(Long.MIN_VALUE / 2 + 178))
+        .ext_rep_sint64(list(Long.MIN_VALUE / 2 + 178))
+        .ext_rep_bool(list(true))
+        .ext_rep_float(list(1.2345e6F))
+        .ext_rep_double(list(1.2345e67))
+        .ext_rep_nested_enum(list(AllTypes.NestedEnum.A))
+        .ext_rep_nested_message(list(nestedMessage))
+        .ext_pack_int32(list(Integer.MAX_VALUE))
+        .ext_pack_uint64(list(Long.MIN_VALUE / 2 + 178))
+        .ext_pack_sint64(list(Long.MIN_VALUE / 2 + 178))
+        .ext_pack_bool(list(true))
+        .ext_pack_float(list(1.2345e6F))
+        .ext_pack_double(list(1.2345e67))
+        .ext_pack_nested_enum(list(AllTypes.NestedEnum.A))
+        .ext_map_int32_int32(singletonMap(1, 2))
+        .ext_map_string_string(singletonMap("key", "value"))
+        .ext_map_string_message(singletonMap("message", new AllTypes.NestedMessage(1)))
+        .ext_map_string_enum(singletonMap("enum", AllTypes.NestedEnum.A));
+  }
+
+  private final Moshi moshi = new Moshi.Builder()
+      .add(new WireJsonAdapterFactory())
+      .build();
+
+  @Test public void serializationOfAllTypes() {
+    AllTypes allTypes = createBuilder().build();
+    String json = moshi.adapter(AllTypes.class).toJson(allTypes);
+    assertThat(json).isEqualTo(JSON);
+  }
+
+  @SuppressWarnings("ConstantConditions")
+  @Test public void deserializationOfAllTypes() throws IOException {
+    JsonAdapter<AllTypes> allTypesAdapter = moshi.adapter(AllTypes.class);
+
+    AllTypes allTypes = createBuilder().build();
+    AllTypes parsed = allTypesAdapter.fromJson(JSON);
+    assertThat(parsed).isEqualTo(allTypes);
+    assertThat(parsed.toString()).isEqualTo(allTypes.toString());
+    assertThat(allTypesAdapter.toJson(parsed)).isEqualTo(allTypesAdapter.toJson(allTypes));
+  }
+
+  @Test public void omitsUnknownFields() {
+    AllTypes.Builder builder = createBuilder();
+    builder.addUnknownField(9000, FieldEncoding.FIXED32, 9000);
+    builder.addUnknownField(9001, FieldEncoding.FIXED64, 9001L);
+    builder.addUnknownField(9002, FieldEncoding.LENGTH_DELIMITED,
+        ByteString.of((byte) '9', (byte) '0', (byte) '0', (byte) '2'));
+    builder.addUnknownField(9003, FieldEncoding.VARINT, 9003L);
+
+    AllTypes allTypes = builder.build();
+    String json = moshi.adapter(AllTypes.class).toJson(allTypes);
+    assertThat(json).isEqualTo(JSON);
+  }
+
+  @SuppressWarnings("ConstantConditions")
+  @Test public void nullRepeatedField() throws IOException {
+    RepeatedPackedAndMap parsed = moshi.adapter(RepeatedPackedAndMap.class).lenient()
+        .fromJson("{rep_int32=null,pack_int32=null,map_int32_int32=null}");
```

Comment:
The test is ported from `wire-gson-support`, not sure if it was better to use `lenient` or to rewrite the json here.

Reply:
we want strict please